### PR TITLE
Moving log collector script to Amazon eks ami repo

### PR DESCRIPTION
**Pull request by nithu0115**
*Description of changes:*

Moving the log collector script to this Amazon EKS AMI repository. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mogren** [comment]
I think this needs some updates to work with v1.4.0 and later @nithu0115.

**Comment by nithu0115** [comment]
WIP

**Comment by nithu0115** [comment]
@micahhausler @mogren

File: log-collector-script/eks-log-collector.sh [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+      warning "The current operating system is not supported."
+      ;;
+  esac
+
+  ok
+}
+
+get_ipamd_info() {
+  try "collect L-IPAMD information"
+  if [[ "${ignore_introspection}" == "false" ]]; then
```
Comment (mogren) [inline log-collector-script/eks-log-collector.sh]: Hmm, shouldn't this be checking that it is not true? Currently, it's off by default it seems.

File: log-collector-script/eks-log-collector.sh [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+cleanup() {
+  rm --recursive --force "${COLLECT_DIR}" >/dev/null 2>&1
+}
+
+init() {
+  check_required_utils
+  version_output
+  is_root
+  systemd_check
+  get_pkgtype
```
Comment (mogren) [inline log-collector-script/eks-log-collector.sh]: Can we log the arguments as well?

File: log-collector-script/eks-log-collector.sh [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+  try "collect L-IPAMD information"
+  if [[ "${ignore_introspection}" == "false" ]]; then
+    for entry in ${IPAMD_DATA[*]}; do
+      curl --max-time 3 --silent http://localhost:61679/v1/"${entry}" >> "${COLLECT_DIR}"/ipamd/"${entry}".txt
+    done
+  else
+    echo "Ignoring IPAM introspection stats as mentioned" >> "${COLLECT_DIR}"/ipamd/ipam_introspection_ignore.txt
+  fi
+
+  if [[ "${ignore_metrics}" == "false" ]]; then
```
Comment (mogren) [inline log-collector-script/eks-log-collector.sh]: Same here

File: log-collector-script/eks-log-collector.sh (hunk 1 above)
Comment (nithu0115) [reply log-collector-script/eks-log-collector.sh]: I have set ignore_introspection, ignore_metrics parameters default to [false](https://github.com/awslabs/amazon-eks-ami/blob/f6155ab958b630d7f8a45a13787d0dd2073481a5/log-collector-script/eks-log-collector.sh#L34-L37).

File: log-collector-script/eks-log-collector.sh (hunk 2 above)
Comment (nithu0115) [reply log-collector-script/eks-log-collector.sh]: [Added](https://github.com/awslabs/amazon-eks-ami/blob/f6155ab958b630d7f8a45a13787d0dd2073481a5/log-collector-script/eks-log-collector.sh#L223)

File: log-collector-script/eks-log-collector.sh (hunk 3 above)
Comment (nithu0115) [reply log-collector-script/eks-log-collector.sh]: I have set ignore_introspection, ignore_metrics parameters default to false.

**Pull request merged**